### PR TITLE
Handle async rules in script

### PR DESCRIPTION
**Pull request by booc0mtaco**
- use async/await checks on each rule
- handle promises (fallback to auto auto-resolve)
- use chalk for styling example

### CODE REVIEW

File: dev-doctor.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = [
   {
     type: "exist",
-    glob: "**/*/test.js",
+    glob: "./rules/**/*.spec.js",
```
Comment (booc0mtaco) [inline dev-doctor.config.js:4]: dogfooding :) (tho probably not the kind of use we intend)

File: index.js [hunk 2]
```diff
@@ -61,23 +62,34 @@ const RULE_TYPES = {
...
       const enhancedRule = Object.assign({}, rule, getMeta(type));
 
       spinner.start(enhancedRule.description);
 
       if (rules[type]) {
-        const ruleResult = { ...rule, pass: rules[type](opts) };
-        results.push(ruleResult);
-        spinner[ruleResult.pass ? "succeed" : "fail"]();
+        try {
+          const ruleResult = { ...rule, pass: await rules[type](opts) };
```
Comment (booc0mtaco) [inline index.js:81]: if the rule returns a promise, when we can handle it (and if it doesn't non-promise stuff is treated like `Promise.resolve()`

File: index.js [hunk 3]
```diff
@@ -61,23 +62,34 @@ const RULE_TYPES = {
...
+          spinner[ruleResult.pass ? "succeed" : "fail"]();
+          results.push(ruleResult);
+        } catch (e) {
+          const ruleResult = { ...rule, pass: false };
+
+          spinner.fail();
+          results.push(ruleResult);
```
Comment (booc0mtaco) [inline index.js:88]: Fallback in case the promise rejects for some unknown reason (more to think about here)

**Comment by booc0mtaco** [comment]
Screenshot 2023-02-01 at 19 08 47

**Review by timzchang — APPROVED** [review]

**Pull request merged**